packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js

```diff
@@ -197,6 +193,15 @@ export const useGiftCards = props => {
         setIsCartUpdating
     ]);
 
+    useEffect(() => {
+        // If either operation results in an error, show the error
+        setShouldDisplayCardError(
+            (mostRecentAction === actions.APPLY && applyCardResult.error) ||
+                (mostRecentAction === actions.CHECK_BALANCE &&
+                    balanceResult.error)
+        );
```

Comment:
Do I need to clean up in this effect? Returning `shouldDisplayCardError` to false when the component is unmounted?

Reply:
I guess you are fine with that. Since the component using this will be unmounted, we should be good.

One suggestion would be to convert this to `if-else` condition and then set `setShouldDisplayCardError` in those. It makes it easy to read.